### PR TITLE
PERF: RangeIndex.is_monotonic_inc/dec

### PR DESCRIPTION
**Pull request by sinhrks**
- [x] tests added / passed
- [x] passes `git diff upstream/master | flake8 --diff`
- [x] whatsnew entry

as the same as `is_unique`, it should be monotonic by definition

```
%timeit -n 1 -r 1 pd.RangeIndex(0, 1000000).is_monotonic_increasing

# current master
#1 loop, best of 1: 17.2 ms per loop

# this PR
#1 loop, best of 1: 45.1 µs per loop
```

### CODE REVIEW

File: pandas/tests/indexes/test_range.py [hunk 1]
```diff
@@ -315,6 +315,16 @@ def test_is_monotonic(self):
...
+        index = RangeIndex(2, 1)
+        self.assertTrue(index.is_monotonic)
+        self.assertTrue(index.is_monotonic_increasing)
+        self.assertTrue(index.is_monotonic_decreasing)
+
+        index = RangeIndex(1, 1)
+        self.assertTrue(index.is_monotonic)
+        self.assertTrue(index.is_monotonic_increasing)
+        self.assertTrue(index.is_monotonic_decreasing)
+
```
Comment (jorisvandenbossche) [inline pandas/tests/indexes/test_range.py]: Maybe add a test for the empty case as well?

```
In [23]: pd.RangeIndex().is_monotonic_increasing
Out[23]: True

In [24]: pd.RangeIndex().is_monotonic_decreasing
Out[24]: True
```

Comment (jorisvandenbossche) [inline pandas/tests/indexes/test_range.py]: Although this should maybe rather raise an error (@jreback)

Other index types to raise without arguments, and `range()` does as well.

Comment (sinhrks) [inline pandas/tests/indexes/test_range.py]: @jorisvandenbossche empty case is coverd by ``(1, 1)``.

```
pd.RangeIndex(1, 1).values
# array([], dtype=int64)
```

Let me issue a separate PR to prohibit empty construction.

Comment (jorisvandenbossche) [inline pandas/tests/indexes/test_range.py]: Ah yes, assumed that ``RangeIndex()`` does not has a step yet, but that is of course wrong :-)

Comment (jreback) [inline pandas/tests/indexes/test_range.py]: yeah I don't think ``RangeIndex()`` should be permitted, though I seem to remember needing it for unpickling or somesuch.......

Comment (jreback) [inline pandas/tests/indexes/test_range.py]: let's make a new issue about the empty ``RangeIndex()``

**Pull request closed without merging**